design/ceph/ceph-nvmeof-gateway.md

```diff
@@ -0,0 +1,210 @@
+                                         |     Storage Layer      |
+                                         | +--------------------+ |
+                                         | | OSD 1 (Physical)   | |
+                                         | +--------------------+ |
+                                         +-----------|------------+
+                                                     |
+                                          +----------v----------+
+                                          |   RBD Volume         |
+                                          |   (Block Device)      |
+                                          +----------|------------+
+                                                     |
+                                          +----------v----------+
+                                          | NVMe-oF Target       |
+                                          | (Presents RBD as     |
+                                          | NVMe Namespace)      |
+                                          +----------|------------+
+                                                     |
+                                           +---------v---------+
+                                           |      Network       |
+                                           | TCP/IP or RDMA     |
+                                           | (NVMe-oF Protocol) |
+                                           +---------|-----------+
+                                                     |
+                                      +--------------v--------------+
+                                      |        Client Node          |
+                                      +-----------------------------+
+                                      | NVMe-oF Initiator (nvme-cli)|
+                                      |                             |
+                                      |  Application sees           |
+                                      |  /dev/nvmeXnY               |
+                                      +-----------------------------+
+
+
+
+
+
+* Note:  In this document, when we refer to 'NVMe namespace,' we mean the NVMe protocol's concept of a logical block device, not Kubernetes resource namespaces."
+
+### Components Responsibilities
+
+* **Volume Provisioning**: CSI driver creates RBD image and configures it in NVMe-oF gateway.
+* **Gateway Management**: Rook operator manages gateway lifecycle, scaling, and health.
+* **Load Balancing/k8s Service**: Service provides single endpoint for multiple gateway instances, NVMe namespace exposed over TCP/IP fabric with port 5500.
+* **Client Access**: External clients connect via NVMe-oF initiator to access volumes.
+
+## CephNVMeOFGateway CRD
+
+### Gateway Configuration
+
+- **Group**: `gateway-group-1` - Logical grouping of gateways
+- **Replicas**: `2` - Two gateway instances for HA
+- **Port**: `5500` - NVMe-oF service port
```

Comment:
Does the port need to be defined on the new CR? I assume Rook is not defining the load balancer endpoint, as it is commonly configured outside rook, depending on the environment. The load balancer endpoint likely could always forward to port 5500 on the pods, and no need to make it customizable.